pandas/core/tools/datetimes.py

```diff
@@ -872,13 +872,17 @@ def to_datetime(
         infer_datetime_format=infer_datetime_format,
     )
 
+    result: Timestamp | NaTType | Series | Index
+
     if isinstance(arg, Timestamp):
         result = arg
         if tz is not None:
             if arg.tz is not None:
-                result = result.tz_convert(tz)
+                # error: Too many arguments for "tz_convert" of "NaTType"
+                result = result.tz_convert(tz)  # type: ignore[call-arg]
             else:
-                result = result.tz_localize(tz)
+                # error: Too many arguments for "tz_localize" of "NaTType"
+                result = result.tz_localize(tz)  # type: ignore[call-arg]
```

Comment:
if you update 

```
    def tz_convert(self) -> NaTType: ...
    def tz_localize(self) -> NaTType: ...
```

in `pandas/_libs/tslibs/nattype.pyi` to match the signatures added here in `pandas/_libs/tslibs/timestamps.pyi` you could remove these ignores.

since these ignores appear to be false positives...

```
>>> pd.NaT.tz_convert("utc")
NaT
>>> 
>>> pd.NaT.tz_localize("Europe/London")
NaT
```